packages/block-library/src/widget-area/edit/index.js

```diff
@@ -20,17 +21,43 @@ export default function WidgetAreaEdit( {
 		[ clientId ]
 	);
 
+	const [ isOpened, setIsOpened ] = useState( index === 0 );
+
+	const toggleChildren = () => {
+		setIsOpened( ! isOpened );
+	};
+
 	return (
 		<Panel className={ className }>
-			<PanelBody title={ name } initialOpen={ index === 0 }>
+			<PanelBody
+				title={ name }
+				onToggle={ toggleChildren }
+				opened={ true }
+				className={ isOpened ? 'widget-area-is-opened' : '' }
+			>
 				<EntityProvider
 					kind="root"
 					type="postType"
 					id={ `widget-area-${ id }` }
 				>
-					<WidgetAreaInnerBlocks />
+					<InnerBlocksContainer isOpened={ isOpened } />
 				</EntityProvider>
 			</PanelBody>
 		</Panel>
 	);
 }
+
+function InnerBlocksContainer( { isOpened } ) {
+	const props = isOpened
+		? {}
+		: {
```

Comment:
Maybe we should add a `TODO:` comment here in case we want to extract it into a component or a module.

Reply:
I'm not sure if linter will like it, and besides I'd like to make that cleanup an immediate follow-up to this PR. It seems like the widgets editor is the only place where we need to hide chunks of DOM so adding a component into that package should be reasonable. This would also mean we also have to move the widget-area block to edit-widgets.

Reply:
Sounds good to me!